editor/src/components/assets.ts

```diff
@@ -56,6 +63,12 @@ export function getProjectContentsChecksums(tree: ProjectContentTreeRoot): Githu
       checksums[filename] = getSHA1Checksum(file.fileContents.code)
     } else if (isAssetFile(file) && file.base64 != undefined) {
       checksums[filename] = getSHA1Checksum(file.base64)
+    } else if (isImageFile(file)) {
+      if (file.gitBlobSha) {
+        checksums[filename] = file.gitBlobSha
+      } else if (Object.keys(assetChecksums).includes(filename)) {
+        checksums[filename] = assetChecksums[filename]
+      }
```

Comment:
I think this should become exhaustive and explicitly checked if possible (even if we do nothing in certain cases) so that the type checker can warn us if we introduce any new file types.

Reply:
https://github.com/concrete-utopia/utopia/pull/2933/commits/3798a564ed4d59fce7c7275e4c16fad11dd2a678